ITKImageProcessingFilters/ITKImageWriter.cpp

```diff
@@ -407,15 +399,8 @@ void ITKImageWriter::execute()
   dc->setGeometry(imageGeom);
   
   IDataArray::Pointer currentData = attributeMatrix->getAttributeArray(path.getDataArrayName());
-  QVector<size_t> cDims = currentData->getComponentDimensions();
-  if(cDims.size() > 1)
-  {
-    setErrorCondition(-21019);
-    QString errorMessage = "ITKImageWriter only works on Scalar or Vector data. The selected data array has more than 1 dimension.";
-    notifyErrorMessage(getHumanLabel(), errorMessage.arg(path.getDataArrayName()), getErrorCondition());
-    return;
-  }
-  size_t nComp = cDims[0];
+  size_t nComp = currentData->getNumberOfComponents();
+  QVector<size_t> cDims(1, currentData->getNumberOfComponents());
```

Comment:
This line could be written as:

        Vector<size_t> cDims = { currentData->getNumberOfComponents() }; 

which I think will work with all of the compilers. The syntax is in the C++11 version and Qt just started allowing it around Qt 5.9 I think.

Another question is that what does ITK mean by "Vector"? Is that a 3/4 component or can we have something that is 60x80 or 1x4800? Have to ask the ITK guys for that one probably.

Reply:
Ok, Then it looks like we have the code figured out and the complete patch just needs to be pushed. Don't forget to run git-clang-format before you submit

Reply:
The compiler gives an error about a narrow type conversion if you use this format. It doesn't give an error if you use nComp instead of currentData->getNumberOfComponents().

Reply:
This is because the method is returning an int, specifically a signed 32 bit integer, where as a size_t is an unsigned 64 bit integer. So to quiet the warning and to tell the compiler you are sure you understand try the following:

    QVector<size_t> cDims = { static_cast<size_t>(currentData->getNumberOfComponents()) };